Makefile

```diff
@@ -2,8 +2,6 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-ENSURE_GARDENER_MOD := $(shell go get github.com/gardener/gardener@$$(go list -m -f "{{.Version}}" github.com/gardener/gardener))
-GARDENER_HACK_DIR   := $(shell go list -m -f "{{.Dir}}" github.com/gardener/gardener)/hack
 VERSION             := $(shell cat VERSION)
 REPO_ROOT           := $(shell dirname "$(realpath $(lastword $(MAKEFILE_LIST)))")
 HACK_DIR            := $(REPO_ROOT)/hack
```

Comment:
I think we can get rid of this `HACK_DIR` variable now that there is no `GARDENER_HACK_DIR`. All "hack" related files/binaries would live in `hack` anyway.

Hardcoding `hack` in the file paths might not seem ideal, but in my opinion it's pretty unlikely this directory would ever be renamed, and having full paths to files like `./hack/run-script.sh` would enable us to directly jump to this file in any text editor which provides the capability to jump to a file with a path; instead of having to search the file tree or type the file name in a command pallette. I personally use this feature of my text editor a lot while navigating.

We can re-evaluate other such parts of the Makefile which could be improved to enhance developer productivity.